diff -r 9495708d9b9d cmake/Modules/FindOptiX.cmake
--- a/cmake/Modules/FindOptiX.cmake	Fri Dec 06 21:12:14 2019 +0800
+++ b/cmake/Modules/FindOptiX.cmake	Wed Jul 01 04:34:37 2020 +0000
@@ -39,6 +39,10 @@
   set(bit_dest "")
 endif()

+if (DEFINED ENV{OptiX_INSTALL_DIR})
+  set(OptiX_INSTALL_DIR $ENV{OptiX_INSTALL_DIR})
+endif()
+
 macro(OPTIX_find_api_library name version)
   find_library(${name}_LIBRARY
     NAMES ${name}.${version} ${name}
diff -r 9495708d9b9d cmake/Modules/FindOpticksGLEW.cmake
--- a/cmake/Modules/FindOpticksGLEW.cmake	Fri Dec 06 21:12:14 2019 +0800
+++ b/cmake/Modules/FindOpticksGLEW.cmake	Wed Jul 01 04:34:37 2020 +0000
@@ -17,7 +17,7 @@
 )
 find_library( OpticksGLEW_LIBRARY
               NAMES glew GLEW libglew32 glew32
-              PATHS ${OpticksGLEW_PREFIX}/lib )
+              PATHS ${OpticksGLEW_PREFIX}/lib64 )

 if(OpticksGLEW_VERBOSE)
   message(STATUS "OpticksGLEW_MODULE      : ${OpticksGLEW_MODULE}")
diff -r 9495708d9b9d externals/openmesh.bash
--- a/externals/openmesh.bash	Fri Dec 06 21:12:14 2019 +0800
+++ b/externals/openmesh.bash	Wed Jul 01 04:34:37 2020 +0000
@@ -1090,7 +1090,7 @@

 openmesh-env(){  olocal- ; opticks- ; }
 #openmesh-vers(){ echo 4.1 ; }
-openmesh-vers(){ echo 6.3 ; }
+openmesh-vers(){ echo 7.1 ; }

 openmesh-info(){ cat << EOI

diff -r 9495708d9b9d oglrap/OpticksViz.cc
--- a/oglrap/OpticksViz.cc	Fri Dec 06 21:12:14 2019 +0800
+++ b/oglrap/OpticksViz.cc	Wed Jul 01 04:34:37 2020 +0000
@@ -533,11 +533,11 @@

 void OpticksViz::renderLoop()
 {
-    if(m_interactivity == 0 )
-    {
-        LOG(LEVEL) << "early exit due to InteractivityLevel 0  " ;
-        return ;
-    }
+   // if(m_interactivity == 0 )
+   // {
+   //     LOG(LEVEL) << "early exit due to InteractivityLevel 0  " ;
+   //     return ;
+   // }
     LOG(LEVEL) << "enter runloop ";

     //m_frame->toggleFullscreen(true); causing blankscreen then segv
diff -r 9495708d9b9d optickscore/OpticksMode.cc
--- a/optickscore/OpticksMode.cc	Fri Dec 06 21:12:14 2019 +0800
+++ b/optickscore/OpticksMode.cc	Wed Jul 01 04:34:37 2020 +0000
@@ -98,15 +98,15 @@
     m_noviz(ok->hasArg(NOVIZ_ARG_)),
     m_forced_compute(false)
 {
-    if(SSys::IsRemoteSession())
-    {
-        m_mode = COMPUTE_MODE ;
-        m_forced_compute = true ;
-    }
-    else
-    {
+   // if(SSys::IsRemoteSession())
+   // {
+   //     m_mode = COMPUTE_MODE ;
+   //     m_forced_compute = true ;
+   // }
+   // else
+   // {
         m_mode = m_compute_requested ? COMPUTE_MODE : INTEROP_MODE ;
-    }
+   // }
 }

 void OpticksMode::setOverride(unsigned int mode)
diff -r 9495708d9b9d opticksgeo/OpticksHub.cc
--- a/opticksgeo/OpticksHub.cc	Fri Dec 06 21:12:14 2019 +0800
+++ b/opticksgeo/OpticksHub.cc	Wed Jul 01 04:34:37 2020 +0000
@@ -326,8 +326,8 @@
     //assert( m_ok->isTracer() ) ;


-    bool compute = m_ok->isCompute();
-    bool compute_opt = hasOpt("compute") ;
+    bool compute = false; //m_ok->isCompute();
+    bool compute_opt = true; //hasOpt("compute") ;
     if(compute && !compute_opt)
         LOG(error) << "FORCED COMPUTE MODE : as remote session detected " ;